langstream-core/src/main/java/ai/langstream/impl/resources/datasource/ElasticSearchDatasourceConfig.java

```diff
@@ -0,0 +1,91 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ai.langstream.impl.resources.datasource;
+
+import ai.langstream.api.doc.ConfigProperty;
+import ai.langstream.api.doc.ResourceConfig;
+import ai.langstream.api.model.Resource;
+import ai.langstream.impl.resources.BaseDataSourceResourceProvider;
+import ai.langstream.impl.uti.ClassConfigValidator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+
+@Data
+@ResourceConfig(
+        name = "OpenSearch",
+        description = "Connect to Elastic service or ElasticSearch service.")
+public class ElasticSearchDatasourceConfig extends BaseDatasourceConfig {
+
+    public static final BaseDataSourceResourceProvider.DatasourceConfig CONFIG =
+            new BaseDataSourceResourceProvider.DatasourceConfig() {
+
+                @Override
+                public Class getResourceConfigModelClass() {
+                    return ElasticSearchDatasourceConfig.class;
+                }
+
+                @Override
+                public void validate(Resource resource) {
+                    ClassConfigValidator.validateResourceModelFromClass(
+                            resource,
+                            ElasticSearchDatasourceConfig.class,
+                            resource.configuration(),
+                            false);
+                }
+            };
+
+    @ConfigProperty(
+            description =
+                    """
+                    Whether to use https or not.
+                    """,
+            defaultValue = "true")
+    private boolean https = true;
+
+    @ConfigProperty(
+            description =
+                    """
+                    Host parameter for connecting to OpenSearch.
+                    Valid both for OpenSearch and AWS OpenSearch Service/Serverless.
+                    """,
+            required = true)
+    private String host;
+
+    @ConfigProperty(
+            description =
+                    """
+                    Port parameter for connecting to OpenSearch service.
+                    """,
+            defaultValue = "9200")
+    private int port;
+
+    @ConfigProperty(
+            description = """
+                    Api Key.
+                    """,
+            required = false)
+    @JsonProperty("api-key")
+    private String apiKey;
+
+    @ConfigProperty(
+            description =
+                    """
+                    Name of the index to to connect to.
+                    """,
+            required = true)
+    @JsonProperty("index-name")
+    private String indexName;
+}
```

Comment:
In our experiments feature, we need to write to multiple indexes, one for each embedding model. Ideally, we'd be able to configure the index on the agent, not in the data source. That way we don't need a datasource for each embedding model (like we do for Pinecone today). Even better than that, we would be able to read a field from the record to determine which data source to write to/query. That way, we don't need to have an agent for each embedding model. The Astra DB connectors works that way. We determine the collection from the record, so we only need one agent for writer/reader,  simplifying the setup. 

The feasibility of this will depend on how the ES client is set up. If the ES client can only work with one index at a time, then have a single writer/reader agent won't be possible. But if we can pass the index name when writing/reading then it likely could be made to work.